modin/core/storage_formats/pandas/utils.py

```diff
@@ -18,73 +18,36 @@
 import pandas
 
 
-def get_default_chunksize(length, num_splits):
+def compute_chunksize(index_len, num_splits, default_block_size=None):
     """
-    Get the most equal chunksize possible based on length and number of splits.
-
-    Parameters
-    ----------
-    length : int
-        The length to split (number of rows/columns).
-    num_splits : int
-        The number of splits.
-
-    Returns
-    -------
-    int
-        Computed chunksize.
-    """
-    return (
-        length // num_splits if length % num_splits == 0 else length // num_splits + 1
-    )
+    Compute the number of elemenents (rows/columns) to include in each partition.
 
-
-def compute_chunksize(
-    df,
-    num_splits,
-    default_block_size=None,
-    axis=None,
-):
-    """
-    Compute the number of rows and/or columns to include in each partition.
+    Chunksize is defined the same for the axes.
 
     Parameters
     ----------
-    df : pandas.DataFrame
-        DataFrame to split.
+    index_len : int
+        Element count in a index.
     num_splits : int
-        Number of splits to separate the DataFrame into.
+        The number of splits.
     default_block_size : int, optional
         Minimum number of rows/columns in a single split.
         If not specified, the value is assumed equal to ``MinPartitionSize``.
-    axis : int, optional
-        Axis to split across. If not specified - split accros both axes.
 
     Returns
     -------
-    int if axis was specified, tuple of ints otherwise,
-        If axis is 1 or 0, returns an integer number of rows/columns to split the
-        DataFrame. If axis is None, returns a tuple containing both.
+    int
+        Integer number of rows/columns to split the DataFrame will be returned.
     """
-    default_block_size = (
-        MinPartitionSize.get() if default_block_size is None else default_block_size
-    )
-    if axis == 0 or axis is None:
-        row_chunksize = get_default_chunksize(len(df.index), num_splits)
-        # Take the min of the default and the memory-usage chunksize first to avoid a
-        # large amount of small partitions.
-        row_chunksize = max(1, row_chunksize, default_block_size)
-        if axis == 0:
-            return row_chunksize
-    # We always execute this because we can only get here if axis is 1 or None.
-    col_chunksize = get_default_chunksize(len(df.columns), num_splits)
+    if default_block_size is None:
+        default_block_size = MinPartitionSize.get()
+
+    chunksize = index_len // num_splits
+    if index_len % num_splits == 0:
+        chunksize += 1
     # Take the min of the default and the memory-usage chunksize first to avoid a
     # large amount of small partitions.
-    col_chunksize = max(1, col_chunksize, default_block_size)
-    if axis == 1:
-        return col_chunksize
-
-    return row_chunksize, col_chunksize
+    return max(1, chunksize, default_block_size)
```

Comment:
this looks weird - comment says "take the min" but code uses `max()`... also why do we pass `1` to `max()`?..

Reply:
Updated